### PR TITLE
JBPM-6204: Extend CaseServiceClient tests for adding dynamic tasks to stage

### PR DESCRIPTION
**Pull request by domhanak**
Added basic tests to check addDynamicTaskToStage error handling.
Refactored whole class to use AssertJ. 
Note that by doing this I found out that some tests that checked  bad container id handling,
were passing even thought the exception was not the one expected.

Let me know if this is OK, I will add ignores to tests or we can simply wait until there is a fix ready for this.
https://issues.jboss.org/browse/JBPM-6223

### CODE REVIEW

**Comment by mswiderski** [comment]
@markcoble seems like JBPM-6223 is assigned to you, any plans on resolving it?

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/jbpm/cases/CaseServiceIntegrationTest.java [hunk 1]
```diff
@@ -75,15 +84,15 @@ protected void addExtraCustomClasses(Map<String, Class<?>> extraClasses) throws
...
+        Assertions.assertThat(caseId).isNotNull();
+        Assertions.assertThat(caseId).startsWith(CLAIM_CASE_ID_PREFIX);
 
         CaseInstance caseInstance = caseClient.getCaseInstance(CONTAINER_ID, caseId);
         assertCarInsuranceCaseInstance(caseInstance, caseId, USER_YODA);
 
         // since roles were not assigned to any users/groups no tasks are available
         List<TaskSummary> tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
-        assertEquals(0, tasks.size());
+        Assertions.assertThat(tasks.size()).isEqualTo(0);
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/jbpm/cases/CaseServiceIntegrationTest.java]: Could be replaced by Assertions.assertThat(tasks).isEmpty();

Comment (domhanak) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/jbpm/cases/CaseServiceIntegrationTest.java]: True, missed this one, thanks.

**Comment by mswiderski** [comment]
this is replaced by https://github.com/kiegroup/droolsjbpm-integration/pull/1055 which contains fixes

**Pull request closed without merging**